src/posix.mak

```diff
@@ -511,14 +511,40 @@ zip:
 ../changelog.html: ../changelog.dd $(HOST_DMD_PATH)
 	CC=$(HOST_CXX) $(HOST_DMD_RUN) -Df$@ $<
 
-#############################
-
-ifneq ($(DOCDIR),)
-html: $(DOCDIR)/.generated
-$(DOCDIR)/.generated: $(DMD_SRCS) $(ROOT_SRCS) $(HOST_DMD_PATH) project.ddoc
-	$(HOST_DMD_RUN) -of- $(MODEL_FLAG) -J$G -J../res -c -Dd$(DOCDIR)\
-	  $(DFLAGS) project.ddoc $(DOCFMT) $(DMD_SRCS) $(ROOT_SRCS)
-	touch $@
+################################################################################
+# DDoc documentation generation
+################################################################################
+
+# BEGIN fallbacks for old variable names
+# should be removed after https://github.com/dlang/dlang.org/pull/1581
+# has been pulled
+DOCSRC?=../dlang.org
+STDDOC?=$(DOCFMT)
+DOC_OUTPUT_DIR?=$(DOCDIR)
+# END fallbacks
+
+# DDoc html generation - this is very similar to the way the documentation for
+# Phobos is built
+ifneq ($(DOCSRC),)
+
+# list all files for which documentation should be generated
+SRC_DOCUMENTABLES = $(ROOT_SRCS) $(DMD_SRCS)
+
+D2HTML=$(foreach p,$1,$(if $(subst package.d,,$(notdir $p)),$(subst /,_,$(subst .d,.html,$p)),$(subst /,_,$(subst /package.d,.html,$p))))
+HTMLS=$(addprefix $(DOC_OUTPUT_DIR)/, \
+	$(call D2HTML, $(SRC_DOCUMENTABLES)))
+
+# For each module, define a rule e.g.:
+# ../web/phobos/ddmd_mars.html : ddmd/mars.d $(STDDOC) ; ...
+$(foreach p,$(SRC_DOCUMENTABLES),$(eval \
+$(DOC_OUTPUT_DIR)/$(call D2HTML,$p) : $p $(STDDOC) ;\
+  $(HOST_DMD_RUN) -of- $(MODEL_FLAG) -J$G -J../res -c -Dd$(DOCSRC) -Iddmd\
+  $(DFLAGS) project.ddoc $(STDDOC) -Df$$@ $$<))
+
+$(DOC_OUTPUT_DIR)/. :
+	mkdir -p $@
```

Comment:
Yah, it does seem to work. Ah, the lure of the Kavorka...

It doesn't work in the sense that the dependency is not checked if you have files in that directory. The only reasonable way to make it work is crazy subtle: https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html

I'd say we use the `$(shell ...)` thing everywhere. It works and is rock solid.